### PR TITLE
Restrict ingestion stats fields

### PR DESCRIPTION
**Pull request by wangch079**
This PR restricts what fields can be updated as ingestion stats.

#### Pre-Review Checklist
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [ ] Considered corresponding documentation changes
- [ ] Contributed any configuration settings changes to the configuration reference

### CODE REVIEW

File: connectors/byoc.py [hunk 1]
```diff
@@ -210,6 +217,12 @@ async def update_metadata(self, ingestion_stats=None, connector_metadata=None):
         if connector_metadata is None:
             connector_metadata = {}
 
+        ingestion_stats = {
+            k: v
+            for (k, v) in ingestion_stats.items()
+            if k in ALLOWED_INGESTION_STATS_KEYS
+        }
+
```
Comment (navarone-feekery) [inline connectors/byoc.py:225]: Might be my lack of knowledge on python standards, but is it possible to have this and the algorithm below to be in the same method so the code is in one place?

File: connectors/byoc.py [hunk 2]
```diff
@@ -29,6 +29,13 @@
 JOB_NOT_FOUND_ERROR = "Couldn't find the job"
 UNKNOWN_ERROR = "unknown error"
 
+ALLOWED_INGESTION_STATS_KEYS = (
+    "indexed_document_count",
+    "indexed_document_volume",
+    "deleted_document_count",
+    "total_document_count",
+)
+
```
[the remark this answers is not in the document]
Comment (wangch079) [reply connectors/byoc.py:38]: These are fields defined in `.elastic-connectors-sync-jobs`: [ref](https://github.com/elastic/connectors-python/blob/main/docs/CONNECTOR_PROTOCOL.md#connector-job-index-elastic-connectors-sync-jobs)

It was raised by @artem-shelkovnikov here https://github.com/elastic/connectors-python/pull/515#discussion_r1150226844 that the method should restrict what can be pass inside `ingestion_stats`.

[the remark this answers is not in the document]
Comment (navarone-feekery) [reply connectors/byoc.py:38]: Thanks!

**Review by artem-shelkovnikov — DISMISSED** [review]
Looks good, but would be indeed nice to make it a reusable method

**Review by navarone-feekery — DISMISSED** [review]

**Review by artem-shelkovnikov — APPROVED** [review]

**Pull request merged**